generic_config_updater/gu_common.py

```diff
@@ -174,3 +188,562 @@ def convert_sonic_yang_patch_to_config_db_patch(self, patch):
+            xpath_tokens.extend(new_xpath_tokens)
+            return xpath_tokens
+
+        # check if it is targetting a child container
+        child_container_model = self._get_model(model.get('container'), path_tokens[token_index+1])
+        if child_container_model:
+            new_xpath_tokens = self._get_xpath_tokens_from_container(child_container_model, token_index+1, path_tokens, config[path_tokens[token_index]])
+            xpath_tokens.extend(new_xpath_tokens)
+            return xpath_tokens
+
+        new_xpath_tokens = self._get_xpath_tokens_from_leaf(model, token_index+1, path_tokens, config[path_tokens[token_index]])
+        xpath_tokens.extend(new_xpath_tokens)
+
+        return xpath_tokens
+
+    def _get_xpath_tokens_from_list(self, model, token_index, path_tokens, config):
+        list_name = model['@name']
+
+        tableKey=path_tokens[token_index]
+        listKeys = model['key']['@value']
+        keyDict = self._extractKey(tableKey, listKeys)
+        keyTokens=[f"[{key}='{keyDict[key]}']" for key in keyDict]
+        item_token = f"{list_name}{''.join(keyTokens)}"
+
+        xpath_tokens = [item_token]
+
+        # if whole list-item is needed i.e. if in the path is not referencing child leaf items
+        # Example:
+        #   path: /VLAN/Vlan1000
+        #   xpath: /sonic-vlan:sonic-vlan/VLAN/VLAN_LIST[name='Vlan1000']
+        if len(path_tokens)-1==token_index:
+            return xpath_tokens
+
+        new_xpath_tokens = self._get_xpath_tokens_from_leaf(model, token_index+1, path_tokens,config[path_tokens[token_index]])
+        xpath_tokens.extend(new_xpath_tokens)
+        return xpath_tokens
+
+    def _get_xpath_tokens_from_leaf(self, model, token_index, path_tokens, config):
+        token = path_tokens[token_index]
+
+        # checking all leaves
+        leaf_model = self._get_model(model.get('leaf'), token)
+        if leaf_model:
+            return [token]
+
+        # checking choice
+        choices = model.get('choice')
+        if choices:
+            for choice in choices:
+                cases = choice['case']
+                for case in cases:
+                    leaf_model = self._get_model(case.get('leaf'), token)
+                    if leaf_model:
+                        return [token]
+
+        # checking leaf-list (i.e. arrays of string, number or bool)
+        leaf_list_model = self._get_model(model.get('leaf-list'), token)
+        if leaf_list_model:
+            # if whole-list is to be returned, just return the token without checking the list items
+            # Example:
+            #   path: /VLAN/Vlan1000/dhcp_servers
+            #   xpath: /sonic-vlan:sonic-vlan/VLAN/VLAN_LIST[name='Vlan1000']/dhcp_servers
+            if len(path_tokens)-1==token_index:
+                return [token]
+            list_config = config[token]
+            value=list_config[int(path_tokens[token_index+1])]
+            return [f"{token}[.='{value}']"]
```

Comment:
>{token}[.='{value} [](http://example.com/codeflow?start=22&length=18)

It's difficult to read, could you add some comment? Is the `.` a mistake?

Reply:
To get a leaf-list instance with the value 'val'
  /module-name:container/leaf-list[.='val']

Source: Check examples in https://netopeer.liberouter.org/doc/libyang/master/html/howto_x_path.html

Added a comment to explain this '.' locally.

Reply:
updated